pkg/kubelet/kubelet_pods.go

```diff
@@ -1341,6 +1341,13 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 	spec := &pod.Spec
 	allStatus := append(append([]v1.ContainerStatus{}, s.ContainerStatuses...), s.InitContainerStatuses...)
 	s.Phase = getPhase(spec, allStatus)
+	// When restartPolicy is never and sandbox is exited, we should set pod phase to Failed
```

Comment:
Nit: Instead of a comment, could we do something like...

```
sandboxExited := len(podStatus.SanboxStatuses) !=  0 && len(podStatus.ContainerStatuses)
if sandboxExited && pod.Spec.RestartPolicy == v1.RestartPolicyNever {
...
}
```